text/0000-buildpack-registry-search-api.md

```diff
@@ -0,0 +1,172 @@
+  - Customers looking to quickly indentify usefulful buildpacks to build their applications.
+- What is the expected outcome?  
+  - Customers will be able to quickly find relevant buildpacks for their use cases.
+
+# What it is
+[what-it-is]: #what-it-is
+
+An external API service that exposes endpoints for retrieving buildpack metadata.  Initially, I envision this service providing a few GET endpoints to support plain text query searches, and individual buildpack data retrieval. 
+
+### Supported Endpoints
+
+- **GET /buildpacks**
+  
+  Retrives all the buildpacks in the registry
+
+- **GET /buildpacks?query=text**
+
+  Retrievs all the buildpacks that statisfy the search query.  For example:
+  ```
+  $ curl https://glacial-plateau-95350.herokuapp.com/buildpacks?query=projectriff
+  ```
+  ```json
+  [
+    {
+      "ns":"projectriff",
+      "name":"command-function",
+      "version":"1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"
+    },
+    {
+      "ns":"projectriff",
+      "name":"java-function",
+      "version":"1.4.1",
+      "yanked":false,"addr":"gcr.io/projectriff/java-function@sha256:5eabea8f7b2c09074ec196fe0c321006fb5ad8f282cc918520286d8a0007196f"},
+    {
+      "ns":"projectriff",
+      "name":"node-function",
+      "version":"1.4.1",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/node-function@sha256:194298b826c15bb079c59aed99968d7678a6e1f7a882c9d7f61811e0990717ba"},
+    {
+      "ns":"projectriff",
+      "name":"streaming-http-adapter",
+      "version":"1.4.0",
+      "yanked":false,
+      "addr":"gcr.io/projectriff/streaming-http-adapter@sha256:b202d9ec203e882ee7e3c599d9e867617f909c8b6123e4ce942af47db6e58c45"
+    }
+  ]
+  ```
+
+- **GET /buildpacks/:id**
+
+  Retrieves metadata for a specific buildpack.  This will contain more detailed data.
+  ```
+  $ curl https://glacial-plateau-95350.herokuapp.com/buildpacks/1
+  ```
+  ```json
+  {
+      "description": "The Command Function Buildpack is a Cloud Native Buildpack V3 that provides riff Command Function Invoker to functions",
+      "license": "MIT",
+      "ns":"projectriff",
+      "name":"command-function",
+      "latest_version": "1.4.1",
+      "versions": "/buildpacks/1/versions",
+      "weekly_downloads": 8993,
+      "issues": 44,
+      "addr":"gcr.io/projectriff/command-function@sha256:99f9054abb73635a9b251b61d3627a8ff86508c767f9d691c426d45e8758596f"
+  }
+  ```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+In the initial implementation, we can take advantage of the existing [registry](https://github.com/buildpacks/registry-index) repository to pull buildpack data from.  Pack already performs a clone/pull on this repo for its registry cache.  To keep ensure up-to-date buildpack data, we'd have a polling loop that performs a `git pull`.  This polling would take place in a separate background process.  
```

Comment:
I don't see how pack's mention here ties to this RFC/section.

Reply:
It doesn't really.  I think it probably got sucked into my conscience thought because we're already doing I was thinking about how Pack also does a direct clone of the registry repo.  I can remove it.